### PR TITLE
feat: add support to bind in Display

### LINKED ISSUE #1063: Add support to bind in display
Part of #794 

## Use case

The user needs to show/hide components in the tree.

## Proposal

Use the display option under style to show/hide components.
We need to make the display option bindable to change its value using context.

### PR DESCRIPTION
**Pull request by uziasferreirazup**
### Related Issues

Closes #1063 

### Description and Example

Add support to bind in display

### Checklist

Please, check if these important points are met using `[x]`:

- [X] I read the [PR Guide] and followed the process outlined there for submitting this PR.
- [X] I avoided _breaking changes_ by not changing public APIs that people rely on. 
- [X] I am willing to follow-up on review comments in a timely manner.

[PR Guide]: https://github.com/ZupIT/beagle/blob/master/doc/contributing/pull_requests.md

### CODE REVIEW

**Comment by botbeaglezup** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Big PR</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: android/beagle/src/main/java/br/com/zup/beagle/android/view/custom/BeagleFlexView.kt [hunk 1]
```diff
@@ -47,8 +53,9 @@ internal open class BeagleFlexView(
     var listenerOnViewDetachedFromWindow: (() -> Unit)? = null
 
     fun addView(child: View, style: Style) {
-
-        super.addView(child, flexMapper.makeYogaNode(style))
+        val childYogaNode = flexMapper.makeYogaNode(style)
+        observeStyleChanges(style, this, childYogaNode)
+        super.addView(child, childYogaNode)
```
Comment (matheusribeirozup) [inline android/beagle/src/main/java/br/com/zup/beagle/android/view/custom/BeagleFlexView.kt:58]: Can you extract this code to a function?

File: android/beagle/src/main/java/br/com/zup/beagle/android/engine/mapper/FlexMapper.kt [hunk 2]
```diff
@@ -40,16 +43,32 @@ class FlexMapper {
...
         alignContent = makeYogaAlignContent(style.flex?.alignContent) ?: YogaAlign.FLEX_START
-        if(style.flex?.flex == null) {
+        if (style.flex?.flex == null) {
             flexGrow = style.flex?.grow?.toFloat() ?: 0.0f
             flexShrink = style.flex?.shrink?.toFloat() ?: 1.0f
         }
         style.flex?.flex?.toFloat()?.let { flex = it }
-        display = makeYogaDisplay(style.display) ?: YogaDisplay.FLEX
+
+        display = YogaDisplay.FLEX
```
Comment (matheusribeirozup) [inline android/beagle/src/main/java/br/com/zup/beagle/android/engine/mapper/FlexMapper.kt:52]: Is this change necessary? If so, you must remove `makeYogaDisplay` method and his enums from source code

**Review by matheusribeirozup — APPROVED** [review]

**Review by matheusribeirozup — APPROVED** [review]

**Comment by botbeaglezup** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Error">
          1 Error
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:no_entry_sign:</td>
      <td data-sticky="false">Tests have failed, see below for more information.</td>
    </tr>
  </tbody>
</table>

### Tests: 

Name | Classname | Time|
--- | --- | ---|
GIVEN display NONE WHEN call observe bind changes THEN it should set in yoga node correct display | br.com.zup.beagle.android.engine.mapper.FlexMapperTest | 0.584|
GIVEN display FLEX WHEN call observe bind changes THEN it should set in yoga node correct display | br.com.zup.beagle.android.engine.mapper.FlexMapperTest | 0.55|


<p align="right" data-meta="generated_by_android">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Review by matheusribeirozup — APPROVED** [review]

File: android/beagle/src/test/java/br/com/zup/beagle/android/view/custom/BeagleFlexViewTest.kt [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+            rootView = rootViewMock,
+            style = styleMock,
+            flexMapper = flexMapperMock,
+            viewRendererFactory = viewRendererFactoryMock,
+            viewModel = screenContextViewModelMock
+        )
+
+        // Then
+        verify {
+            flexMapperMock.observeBindChangesFlex(any(), rootViewMock, beagleFlexView, beagleFlexView.yogaNode)
```
Comment (gabrielleonardozup) [inline android/beagle/src/test/java/br/com/zup/beagle/android/view/custom/BeagleFlexViewTest.kt:57]: Is any() as first argument correct? Or should be styleMock ?

**Review by gabrielleonardozup — COMMENTED** [review]
I made a comment.

**Review by gabrielleonardozup — APPROVED** [review]

**Review by paulomeurerzup — APPROVED** [review]
